### ISSUE TITLE
Recent change in set_location suspected reason for getting stuck in blocks

### ISSUE THREAD

**Issue by 1AmYF** [piqueserver/piqueserver#587]
While running a bot gamemode i noticed several getting stuck in blocks. I tested the same script few month ago without seeing such issues. Also someone else mentioned in the piqueserver chat
>  Sometimes when spawn, I get stuck in blocks. I can not find the reason for this

I suspect this could be the reason: https://github.com/piqueserver/piqueserver/pull/576. The "before" location was considered an error, but i guess there was a specific reason for it to be that way. After reverting my checkout to a version before that commit, the issue did not reappear.

**Comment by Jipok** [comment]
This problem was before my patch.
#576 only affects the call to `set_location` with the `location` argument. Which is almost never happening.
And set_location is in no way called at `spawn`. `spawn` uses the CreatePlayer package. And the coordinates are calculated as follows:
```
x, y, z = self.get_spawn_location()
x += 0.5
y += 0.5
z -= 2.4
```

**Comment by NotAFile** [comment]
The code was clearly incorrect IMO. It stored a different location internally than it sent to the client. So the real question is "which of the two locations is the correct one".

There's no right answer, but we decided that the position sent to the client was the correct one because:

 - The value stored on the server is never visible to users, so most code would have been written oblivious of the fact that that value was incorrect.
 - Indeed, all code using `set_location`, both in core and scripts, intended to set the value sent to the client.
 - In most cases, physics would push the player up onto the block in the next tick.
 - If not, clients overwrite the server position every second, so the incorrect value would not live long anyway.

So unfortunately, this means that the bot code is the thing that must be fixed.

**Comment by Jipok** [comment]
I don't know anything about the bots that he uses. But this is my phrase:
> Sometimes when spawn, I get stuck in blocks. I can not find the reason for this

None of my scripts override any location code or spawn code. But I and my bots sometimes get stuck. It was also up to #576

**Comment by 1AmYF** [comment]
> Unfortunately for you, this means that the bot has to be fixed, not our code.

Alright, fixed it by substracting the 0.5 from x and y before passing it to set_location_safe
